components/channel_layout/center_channel/index.ts

```diff
@@ -83,5 +83,5 @@ const connector = connect(mapStateToProps, mapDispatchToProps);
 
 export type PropsFromRedux = ConnectedProps<typeof connector>;
 
-export default connector(CenterChannel);
+export default withRouter(connector(CenterChannel));
```

Comment:
to access router props. see https://github.com/mattermost/mattermost-webapp/pull/11059#discussion_r962352152